fedora-26: replace @workstation-product-environment by @gnome-desktop

fepitre opened the pull request:
I obtained a template of size ~990mo. Like I did for fc27 the workspace group installs so much useless things. Need to fix dnf also for fc27. I tested it briefly on 3.2 and things seem fine.

rustybird commented:
Hmm, this removed lots of fairly essential tools like less and man; some other common ones like wget, rsync, bzip2, lsof, git; then CUPS, LibreOffice, [etc. etc.](https://gist.githubusercontent.com/rustybird/3ace5f1e2c50d70b48c8adcf803bb1a6/raw/af17f91f084fdc68a821abd957f8f87b01d14f91/pkgs.txt)

Now fedora-26 feels kind of weird, like fedora-26-minimal with random stuff bolted on here and there.

fepitre commented:
In my opinion it goes well with having a less minimal fedora-minimal template.

rustybird commented:
You mean, merge fedora-26 and fedora-26-minimal into a single template whose package selection would be a middle ground between the two current templates? Sounds interesting, but it would be a big change - as is the reduced package selection for fedora-26 at this relatively late stage, rc4.

I suspect that the reason more people haven't yet complained is that many rc4 testers probably just updated their templates from previous release candidates, thereby keeping `@workstation-product-environment`. But e.g. https://github.com/QubesOS/qubes-issues/issues/3580 looks like it could be part of the fallout.

fepitre commented:
I don't mean not to merge them. Fedora default group workstation-product-environment seems to have grown more and more. IMHO, the latest templates of Fedora was uselessly too big. You have probably followed the Marek's topic about Qubes images (https://groups.google.com/d/topic/qubes-devel/pr8EzL7PjqE/discussion) to know what it should be put on the Qubes DVD or not etc.  So, such a big downsizing of the fedora-26+ templates can be seen as an intermediate template between the minimal one and the old big one. We could now add previously included useful tools in the packages list. Do you have a particular list (not necessary related to gnome)? I can do a rebuilt of the template to obtain the size.

rustybird commented:
> Do you have a particular list (not necessary related to gnome)?

It would be a huge list though (of packages that I'm surprised aren't in a "full" template), many dozens from the gist linked above - and include even something large like LibreOffice, probably negating most of the size reduction.

I realize I'm bringing this up pretty late, not having realized the scope of this PR before - but IMO it would be better to mostly stay away from second-guessing Fedora's packaging choices. It's too much effort that might still result in an incoherent-feeling template, and it conflicts with the ["We try to respect each distro’s culture"](https://www.qubes-os.org/faq/#what-is-qubes-attitude-toward-changing-guest-distros) stance.

So my proposal (/cc @marmarek) would be to revert this PR, single-layer DVD fetishists ;) be damned. Which is not to say that reducing ISO bloat isn't a worthwhile goal, but it has to be weighed against UX cost.

marmarek commented:
> include even something large like LibreOffice, probably negating most of the size reduction.

Well, LibreOffice isn't included in default template for [2+ years](https://github.com/QubesOS/qubes-builder-fedora/commit/c11ca333f20919cd3bc5be739a728218c536f56e#diff-703ef067860b0459a53a98776667b972)

Also, looking at your list, there are *a lot* of packages useless in Qubes VM, using not only disk space, but in many cases also RAM. Some examples:
 - gdm
 - avahi (useless behind NAT)
 - openssh-server
 - selinux-policy-targeted
 - xorg-x11-drv-*
 - fedora-workstation-backgrounds
 - gnome-shell-*
 - hyperv*
 - sssd-*

Previously we've maintained exclude list for that and it was bigger and bigger. So, if something really needed for average user is missing, lets add it to the list, instead of pulling a lot of useless stuff.
Also, I'd focus on stuff useful for less advanced users. I'm sure more advanced users are able to install useful packages themselves and many probably have some favorite list of them (completed over previous migrations to new template) anyway. The fact that the first thing I do on new template is installing mutt, pv, w3m, ipython, tcpdump, strace, gdb (and more) doesn't mean I expect them to be installed by default.

This repository also define "fullyloaded" template flavor. We can install everything defined in workstation-product-environment there, and actually upload the template to the repository (but not include on installation iso). Maybe even set larger root volume for it. I think this is better approach, also regarding required bandwidth - better download huge template only when user explicitly want it, instead of forcing _everyone_ to download much bugger image. That DVD size limit, while itself quite arbitrary, I think is also sensible one for this trade-off.

fepitre commented:
I'm also supporting this way of proceeding. I don't want a template with default useless things. We could also consider to use the xfce template as default to be consistent with Qubes dom0 but this would be another discussion.

rustybird commented:
> Well, LibreOffice isn't included in default template for 2+ years

Sorry, right. I thought it was. (Although to me it makes sense that Fedora includes it.)

> I'm sure more advanced users are able to install useful packages themselves and many probably have some favorite list of them (completed over previous migrations to new template) anyway.

Those favorite lists are about to get a lot longer, by trial and error... Maybe I'm too easily annoyed by the experience of `$ something-common` -> `something-common: command not found` -> install in template -> restart VM -> repeat.

> So, if something really needed for average user is missing, lets add it to the list

Don't know who the average Qubes user is - some things that stood out for me:

- `@Common NetworkManager Submodules`
- `@Printing Support`
- bc
- bzip2
- git
- less
- lsof
- man-db, man-pages
- unzip
- wget 
- wireless-tools
- iw

> This repository also define "fullyloaded" template flavor. We can install everything defined in workstation-product-environment there, and actually upload the template to the repository (but not include on installation iso).

Fair enough.

marmarek commented:
Thanks @rustybird for the list! @fepitre could you check resulting template size?

rustybird commented:
Maybe also

- `@Multimedia`
- zip if we're adding unzip
- attr, dos2unix, rsync, tree

fepitre commented:
Sure. I launch the build for F26 and F26+XFCE

fepitre commented:
I brutally added all the packages you suggested:
bc
bzip2
git
less
lsof
man-db, man-pages
zip
unzip
wget
wireless-tools
iw
@common
@multimedia
@printing
@networkmanager-submodules
attr
dos2unix
rsync
tree

and the template is 1.5G! I will retry by removing the group @multimedia and @printing

fepitre commented:
Sorry, I deleted my comment previous comment. I have not updated my repos....I do the rebuild with @gnome-desktop...

fepitre commented:
Almost done. I added successively group packages (last to build is with @ printing). I relaunch build for fc26 et fc26+xfce. For FC26 with all except @ printing, it is almost 1.1G so this is reasonable.

fepitre commented:
@marmarek : here are the results
1095M qubes-template-fedora-26-4.0.0-201802151336.noarch.rpm
1131M qubes-template-fedora-26-xfce-4.0.0-201802151403.noarch.rpm

see the packages list: https://github.com/fepitre/qubes-builder-fedora/commit/10849e95369b66f3068ee0a2b0436fb6ef39b201

I cleaned a little bit the packages file for fc26/27, added into XFCES ones but I wait that you accept the PR "Update XFCE" before cleaning also the XFCE ones.

fepitre commented:
@marmarek: would you like I open a PR?

marmarek commented:
Yes, please.

fepitre commented:
This is strange. Here is what I have done: 
- cleaned everything, used the latest version of the master conf,
- added USE_QUBES_REPO_VERSION=4.0, USE_QUBES_REPO_TESTING=1

With the default TEMPLATE_ROOT_WITH_PARTITIONS = 1, the template size is 1263974972 qubes-template-fedora-26-4.0.0-201802172224.noarch.rpm and with TEMPLATE_ROOT_WITH_PARTITIONS = 0, the template size is 1149483024 qubes-template-fedora-26-4.0.0-201802180748.noarch.rpm

I don't understand why I have smaller images. I will investigate.
